### ISSUE TITLE
Support Trigger and FIFO Buffer reading

### ISSUE THREAD

**Issue by mperino** [adafruit/Adafruit_ADXL375#8]
While I appreciate the goals of the unified simple sensor library in this case the most critical feature of this sensor is left out.
Reading data from the ADXL375 in any kind of loop causes max values to be missed, and really misses out on the functionality built into this lovely little sensor.  
As discussed in the forum:
https://forums.adafruit.com/viewtopic.php?p=1020141

Because it's the main reason one would choose an ADXL375 over other sensors, I would ask that you implement:
 Implement shock detection
 Record detailed acceleration data after a shock event is triggered
 Return maximum shock amplitude after a shock has occurred
 
 TLDR: Setting Shock Event and reading the FIFO buffer.
